BUG: `ndimage`: Numerical regressions in Dask 2025.2.0

### Describe your issue.

These two tests fail after upgrading Dask from 2025.1.0 to 2025.2.0:
```
FAILED scipy/ndimage/tests/test_filters.py::TestNdimageFilters::test_minimum_filter05_overlap[dask.array] - AssertionError: 
Not equal to tolerance rtol=0, atol=1.5e-06

Mismatched elements: 10 / 15 (66.7%)
Max absolute difference among violations: 8
Max relative difference among violations: 0.88888889
 ACTUAL: array([[2, 2, 1, 1, 1],
       [2, 2, 1, 1, 1],
       [5, 3, 3, 1, 1]])
 DESIRED: array([[3, 2, 5, 1, 4],
       [7, 6, 9, 3, 5],
       [5, 8, 3, 7, 1]])
FAILED scipy/ndimage/tests/test_morphology.py::TestNdimageMorphology::test_grey_erosion01_overlap[dask.array] - AssertionError: 
Not equal to tolerance rtol=0, atol=1.5e-06

Mismatched elements: 9 / 15 (60%)
Max absolute difference among violations: 8
Max relative difference among violations: 8.
 ACTUAL: array([[3, 2, 5, 1, 4],
       [7, 6, 9, 3, 5],
       [5, 8, 3, 7, 1]])
 DESIRED: array([[2, 2, 1, 1, 1],
       [2, 3, 1, 3, 1],
       [5, 5, 3, 3, 1]])
= 2 failed, 76053 passed, 30086 skipped, 56664 deselected, 13347 xfailed, 2 xpassed in 1106.81s (0:18:26) =
```

FYI @lucascolley @lithomas1

--- a/scipy/ndimage/tests/test_filters.py
+++ b/scipy/ndimage/tests/test_filters.py
@@ -1607,6 +1607,7 @@ class TestNdimageFilters:
                                               [5, 3, 3, 1, 1]]), output)
 
     @skip_xp_backends("jax.numpy", reason="assignment destination is read-only")
+    @skip_xp_backends("dask.array", reason="assignment destination is read-only")
     def test_minimum_filter05_overlap(self, xp):
         array = xp.asarray([[3, 2, 5, 1, 4],
                             [7, 6, 9, 3, 5],

--- a/scipy/ndimage/tests/test_morphology.py
+++ b/scipy/ndimage/tests/test_morphology.py
@@ -2332,6 +2332,7 @@ class TestNdimageMorphology:
                                               [5, 5, 3, 3, 1]]))
 
     @skip_xp_backends("jax.numpy", reason="output array is read-only.")
+    @skip_xp_backends("dask.array", reason="output array is read-only.")
     @xfail_xp_backends("cupy", reason="https://github.com/cupy/cupy/issues/8398")
     def test_grey_erosion01_overlap(self, xp):
 

